enhancements/aws-pod-identity.md

```diff
@@ -0,0 +1,569 @@
+---
+title: aws-pod-identity 
+authors:
+  - "@sjenning"
+reviewers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+  - "@joelddiaz"
+  - "@dgoodwin"
+  - "@abhinavdahiya"
+  - "@eparis"
+  - "@cuppett"
+  - "@marun"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2020-03-26
+last-updated: 2020-03-26
+status: provisional
+see-also:
+  - ""  
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Support Native (IAM-based) Pod Identity on AWS
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+### ServiceAccountIssuerDiscovery (RESOLVED)
+
+#### Question
+
+Should we enable the [`ServiceAccountIssuerDiscovery` feature gate](https://github.com/kubernetes/enhancements/issues/1393) (disabled by default, alpha 1.18)? 
+
+Pros:
+- Eliminates the need for external hosting of the OIDC discovery and keys documents
+- Eliminates the need for a controller to sync the keys when the bound
+  ServiceAccount signer rotates
+
+Cons:
+- Requires enabling alpha feature
+- Does not work if kube-apiserver is not publicly accessible from AWS STS
+
+The alternative is to use a pubic S3 bucket to host the OIDC discovery endpoint
+
+Pros:
+- Works even with the apiserver is not accessible by AWS STS
+  (disconnected/private cluster)
```

Comment:
Is the door open to seamlessly transition to ServiceAccountDiscovery later once it's out of alpha?

Reply:
I think so.  We just have to detect if the URL in the Authentication CR is empty or our default and switch it to the ServiceAccountDiscovery URL.  If it is not our default (the customer set it purposefully), we would do nothing.  However, it was decided to do the s3 bucket because ServiceAccountDiscovery won't work for private clusters.